internal/access/access_test.go

```diff
@@ -100,28 +95,26 @@ func TestRequireInfraRole_GrantsFromGroupMembership(t *testing.T) {
 	_, err = data.AssignIdentityToGroups(db, user, []string{tomsGroup.Name})
 	assert.NilError(t, err)
 
-	c, _ := gin.CreateTestContext(httptest.NewRecorder())
 	tx := txnForTestCase(t, db)
-	c.Set(RequestContextKey, RequestContext{
+	rCtx := RequestContext{
 		DBTxn:         tx,
 		Authenticated: Authenticated{User: tom},
-	})
-	authDB, err := RequireInfraRole(c, models.InfraAdminRole)
+	}
+	err = IsAuthorized(rCtx, models.InfraAdminRole)
 	assert.ErrorIs(t, err, ErrNotAuthorized)
-	assert.Assert(t, authDB == nil)
 
 	admin := &models.Identity{Model: models.Model{ID: uid.ID(512)}}
 	grant(t, tx, admin, tomsGroup.PolyID(), models.InfraAdminRole, "infra")
 
-	authDB, err = RequireInfraRole(c, models.InfraAdminRole)
+	err = IsAuthorized(rCtx, models.InfraAdminRole)
 	assert.NilError(t, err)
-	assert.Assert(t, authDB != nil)
 }
 
-func TestRequireInfraRole(t *testing.T) {
+// TODO: mergge this with TestIsAuthorized
```

Comment:
Seems like combining this as part of this PR would be relatively straight-forward, so might be worth doing as part of this PR, just a suggestion though

Reply:
As part of merging these two test cases I'd like to review all the test cases to make sure we've got everything covered. I'd like to do this as a follow up to take more time to review the cases.  I've made a note to myself to do this.